packages/native-stack/src/types.tsx

```diff
@@ -342,6 +342,15 @@ export type NativeStackNavigationOptions = {
    * @platform ios
    */
   headerBackButtonMenuEnabled?: boolean;
+  /**
+   * How the back button behaves by default (when not customized). Available on iOS>=14.
+   * The following values are currently supported (they correspond to https://developer.apple.com/documentation/uikit/uinavigationitembackbuttondisplaymode?language=objc):
+   * - "default" – The navigation item attempts to display a specific title, a generic title, or no title for the Back button, depending on the space available.
+   * - "generic" – The navigation item attempts to display a generic title or no title for the Back button, depending on the space available.
+   * - "minimal" – The navigation item displays the Back button indicator instead of a title.
+   * @platform ios
+   */
+  headerBackButtonDisplayMode?: ScreenStackHeaderConfigProps['backButtonDisplayMode'];
```

Comment:
Let's remove "the navigation item" as it's redundant, and the terminology doesn't match anything else in react navigation, so it'll be confusing unless someone iOS navigation system. I'd also remove "for the Back button" since it's implied since this option is for back button from the option name and the description.

I'd also remove the link to UIKit docs. Our options aren't intended to match native APIs 1:1 and we may change them to what feels more natural.

Though these options are still unclear to me,

- I think "icon" clearer than "indicator" - and matched the existing terminology
- Still unsure what "specific title" and "generic title" mean

Generally we need to match the terminology we already use in react navigation and think in terms of users who don't know anything about native iOS APIs.

Reply:
Thanks for the feedback, I also did some more research and improved the explanations.